WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingLaunchStoreViewModel.swift

```diff
@@ -3,41 +3,126 @@ import Yosemite
 
 /// View model for `StoreOnboardingLaunchStoreView`.
 final class StoreOnboardingLaunchStoreViewModel: ObservableObject {
+    /// UI state of the lauch store view.
+    enum State {
+        // Checking current site's plan to check for free trial
+        case checkingSitePlan
+        // Using free trial. Need to purchase plan before publishing site.
+        case needsPlanUpgrade
+        // Ready to publish site.
+        case readyToPublish
+        // Processing launch store request
+        case launchingStore
+    }
+
     let siteURL: URL
 
-    @Published private(set) var isLaunchingStore: Bool = false
+    @Published private(set) var state: State = .checkingSitePlan
     @Published var error: SiteLaunchError?
 
+    lazy var upgradePlanAttributedString: NSAttributedString = {
```

Comment:
Super nit: should we keep this `private(set)` for extra safety?

Reply:
Done in 3a76cf3a2f86b038d9e6b5108859638efeb807dd